core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/rabbitmq.yml

```diff
@@ -1,30 +1,57 @@
 ---
-- name: RabbitMQ | Check version
-  command: rabbitmqctl version
-  register: rabbitmq_version_current
+- name: RabbitMQ | Register version output
+  command: rabbitmq-diagnostics server_version
+  register: rabbitmq_version
   changed_when: false
 
+- name: RabbitMQ | Register cluster status output
+  command: rabbitmqctl cluster_status --formatter json
+  register: rabbitmq_cluster_status
+  changed_when: false
+  run_once: true
+
 - name: RabbitMQ | Include defaults from rabbitmq role
   include_vars:
     file: roles/rabbitmq/defaults/main.yml
 
-- name: RabbitMQ | Set rabbitmq_master fact
+- name: RabbitMQ | Set facts
   set_fact:
     rabbitmq_master: "{{ groups['rabbitmq'][0] }}"
+    rabbitmq_version_current: "{{ rabbitmq_version.stdout_lines[1] }}"
+    is_clustered: "{{ (rabbitmq_cluster_status.stdout | from_json).running_nodes | length > 1 }}"
+
+- name: RabbitMQ | Print variable values
+  debug:
+    msg: "is_clustered: {{ is_clustered }}; rabbitmq_version_current: {{ rabbitmq_version_current }}"
+
+- name: RabbitMQ | Check for upgrade flag file
+  stat:
+    path: "{{ lock_file }}"
+    get_attributes: false
+    get_checksum: false
+    get_mime: false
+  register: lock_file_status
 
 - name: RabbitMQ | Offline upgrade
-  when: rabbitmq_version_current.stdout != versions.general
+  when: lock_file_status.stat.exists or rabbitmq_version_current is version(versions.general, '<')
   block:
+    - name: RabbitMQ | Create upgrade flag file
+      file:
+        path: "{{ lock_file }}"
+        state: touch
```

Comment:
Would be nice to have at least "mode:" here I think? :thinking: Linter will give you a warning without it.

Reply:
Modified file permissions

Reply:
```shell
cd data/common/ansible/playbooks/roles/ && /home/vscode/.local/bin/ansible-lint upgrade
```
works much better :+1::innocent: